src/coll/options.rs

```diff
@@ -163,6 +163,58 @@ pub struct UpdateOptions {
 
 pub type ReplaceOptions = UpdateOptions;
 
+// Option for map-reduce "out" argument.
+#[derive(Clone)]
+pub enum MapReduceOutOption {
+    Action(bson::Document),
+    Collection(String),
+}
+
+// Options for map-reduce queries.
+#[derive(Clone)]
+pub struct MapReduceOptions {
+    pub finalize: Option<Bson>,
```

Comment:
Could we just use `Option<Bson::JavascriptCode>` in finalize here? Since, the [specs](https://docs.mongodb.com/manual/reference/command/mapReduce/#mapreduce-finalize-cmd) said so.